modules/bidwatchAnalyticsAdapter.js

```diff
@@ -22,53 +26,108 @@ function getAdapterNameForAlias(aliasName) {
   return adapterManager.aliasRegistry[aliasName] || aliasName;
 }
 
-function setOriginalBidder(arg) {
+function cleanArgObject(arg, removead) {
+  if (typeof arg['bidderCode'] == 'string') { arg['originalBidder'] = getAdapterNameForAlias(arg['bidderCode']); }
+  if (typeof arg['creativeId'] == 'number') {
+    arg['creativeId'] = arg['creativeId'].toString();
+  }
+  if (removead && typeof arg['ad'] != 'undefined') {
+    arg['ad'] = 'emptied';
+  }
+  if (typeof arg['gdprConsent'] != 'undefined' && typeof arg['gdprConsent']['vendorData'] != 'undefined') {
```

Comment:
IMO it'd be better to explicitly list what to keep rather than what to remove. As more features are added your payload will probably continue to grow, and I can't imagine new, unrecognized fields being too useful on your end. It would also make it less likely that we'd accidentally rename or remove some field that is important to you, because a simple code search will show us what you are depending on.

Reply:
Thanks for the suggestion @dgirardi 
I am no javascript specialist so I often go to the easiest way I know without searching for subtilities. As I have another PR to send very shortly I'll make the suggested changes with this next one.